app/hooks/useCosmos.ts

```diff
@@ -0,0 +1,57 @@
+import { useCallback, useContext, useState } from "react";
+import { WalletState } from "@/utils/interfaces";
+import { GasPrice, SigningStargateClient } from "@cosmjs/stargate";
+
+interface UseCosmosProps {
+  wallet?: WalletState | null;
+}
+
+export function useCosmos({ wallet }: UseCosmosProps) {
+  const chainId = "cosmoshub-4"; // TODO: Receive from chain info.
+  const keplr = wallet?.provider;
+  const rpcUrl = "https://cosmos-rpc.publicnode.com:443";
```

Comment:
Can you define this as an environment variable ? To avoid having to deploy a new version in case the public node has problems.